dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -58,6 +66,24 @@ def register_new_user(user)
     redirect_to new_user_registration_url
   end
 
+  # TODO: figure out how to avoid skipping CSRF verification for Powerschool
+  skip_before_action :verify_authenticity_token, only: :powerschool
+
+  def extract_powerschool_data(auth)
+    # OpenID 2.0 data comes back in a different format compared to most of our other oauth data.
+    args = JSON.parse(auth.extra.response.message.to_json)['args']
```

Comment:
Casting to JSON and parsing is not pretty, but I've had little luck interacting with the OpenID::Message object itself. This could probably use some cleanup eventually.